cartoframes/context.py

```diff
@@ -249,11 +247,63 @@ def read(self, table_name, limit=None, decode_geom=False, shared_user=None):
         result = self.copy_client.copyto_stream(query)
         df = pd.read_csv(result)
 
+        return self._clean_DataFrame_from_carto(df, table_name, schema)
+
+    def _clean_DataFrame_from_carto(self, df, table_name, schema='public', decode_geom=False):
+        """
+        Clean a DataFrame with a dataset from CARTO:
+        - remove the_geom_webmercator
+        - use cartodb_id as DataFrame index
+        - process date columns
+        - decode geom
+
+        :param df: DataFrame with a dataset from CARTO
+        :type df: pandas DataFrame
+        :param table: table name in user's CARTO account.
+        :type table: str
+        :param schema: table schema name in user's CARTO account.
+        :type schema: str
+        :param decode_geom: Decodes CARTO's geometries into a `Shapely <https://github.com/Toblerity/Shapely>`__
+              object that can be used, for example, in `GeoPandas <http://geopandas.org/>`__.
+        :type decode_geom: bool, optional
+
+        :return: DataFrame
+        """
+        if 'the_geom_webmercator' in df.columns:
```

Comment:
Wouldn't be better to just not request the `the_geom_webmercator` column? We'd lighten the response + save the time spent in popping the column from the dataframe.

Reply:
Postgresql doesn't support "column exceptions". We could check the column list with `_get_table_columns` and add the columns to the query. Do you prefer it?

Reply:
In the best case you'll decrease the size of the response almost to the half. If we don't need it I'd just remove it from the query (yes, by selecting the rest of the columns).

In any case we have to do a query to get the list of columns so I guess it's worth it.

Reply:
:+1: